src/main/java/com/google/devtools/build/lib/skylarkbuildapi/ProtoConfigurationApi.java

```diff
@@ -23,7 +25,65 @@
 @SkylarkModule(
     name = "proto",
     category = SkylarkModuleCategory.CONFIGURATION_FRAGMENT,
-    doc = "A configuration fragment representing protocol buffers."
+    doc = "A configuration fragment representing protocol buffers. "
+              + "<p><b>Do not use these fields directly<b>, they are considered an implementation "
+              + "detail and will be removed after migrating Protobuf rules to Starlark.</p>"
+              // TODO(yannic): Link to generated docs of `proto_toolchain`.
+              // https://github.com/bazelbuild/bazel/issues/9203
+              + "<p>Instead, you can access them through proto_toolchain</p>"
+              + "<p>pre class=\"language-python\">\n"
+              + "def _my_rule_impl(ctx):\n"
+              + "    proto_toolchain = ctx.toolchains[\"@rules_proto//proto:toolchain\"]\n"
+              + "\n"
+              + "    # Contains the protoc binary, as specified by `--proto_compiler`.\n"
+              + "    protoc = proto_toolchain.compiler\n"
+              + "\n"
+              + "    # Contains extra args to pass to protoc, as specified by `--protocopt`.\n"
+              + "    compiler_options = proto_toolchain.compiler_options\n"
+              + "\n"
+              + "    # Contains the strict-dependency mode to use for Protocol Buffers\n"
+              + "    # (i.e. 'OFF`, `WARN`, `ERROR`), as specified by `--strict_proto_deps`.\n"
+              + "    strict_deps = proto_toolchain.strict_deps\n"
+              + "\n"
+              + "my_rule = rule(\n"
+              + "    implementation = _my_rule_impl,\n"
+              + "    attrs = {},\n"
+              + "    toolchains = [\n"
+              + "        \"@rules_proto//proto:toolchain\",\n"
+              + "    ],\n"
+              + ")\n"
+              + "</pre></p>"
 )
 public interface ProtoConfigurationApi {
+  @SkylarkCallable(
+      // Must match the value of `_protocopt_key`
+      // in `@rules_proto//proto/private/rules:proto_toolchain.bzl`.
+      name = "protocopt_do_not_use_or_we_will_break_you_without_mercy",
```

Comment:
@hlopko , how about living without the scary "this is not a supported API" suffix?

Reply:
I thought you like these suffices :) For my perspective I don't think this can cause too much trouble if we expose it under short name. And removing the option using an incompatible flag won't be too bad either. So feel free to use the short name Yannic, if you don't feel strongly.